birdhouse/scripts/read-configs.include.sh

```diff
@@ -16,15 +16,26 @@
 #
 #  # Source the script providing function read_configs.
 #  # read_configs uses COMPOSE_DIR to find default.env and env.local.
-#  . $COMPOSE_DIR/read-configs.include.sh
+#  . ${COMPOSE_DIR}/scripts/read-configs.include.sh
```

Comment:
I'd rather not move this `read-configs.include.sh` file to the `scripts/` folder.

To me, this is another entrypoint to the platform (gives the value of various config vars, source other files like the new `logging.include.sh`).  It's at the same level as the `pavics-compose.sh` file, we want it clearly stand-out.

It's not just scripts in the `scripts/` folder that use it.  Scripts under `deployment/` folder using it to, as all scripts from all external repos that we do not have any controls.

It has no `pavics` in its name, why rename it.  This will force all external repos to update as well, fairly back-ward incompatible.

Reply:
I'm not sure, I understand the use case where `read-configs.include.sh` would be used as entrypoint. As its name indicates, it is intended only to be included to provide utility functions used by other scripts, in the same way the new logging provides reporting utilities. Those should all be contained into some "utils" group. 

Ideally, the file structure should probably be more like: 
```
scripts
  pavics-compose.sh
  utils
    logging.include.sh
    read-configs.include.sh
    ....
  deployment
    ...
```

Or even: 
```
scripts
  birdhouse  (+x)
  utils
    logging.include.sh
    read-configs.include.sh
    ....
  run
    pavics-compose.sh
  deployment
    ...
```
With some kind of `birdhouse run ...`, `birdhouse deploy ...` CLI that sources/dispatches operations to relevant scripts.

Any external script (as in outside this repo) that directly sources `read-configs.include.sh` is not using the code the way it is meant to be. External users should not interact with directly with utilities intended to evolve to adapt the scripts with new capabilities.

Reply:
I tend to agree with @fmigneault that we really *shouldn't* be using the `*.include.sh` scripts as entrypoints. 

@tlvu do you have an example where you're currently using it as an entrypoint now? If you show us the example I'm sure we can help you figure out a more robust way of accomplishing the task without having to use `read-configs.include.sh` as a workaround.

Reply:
By entrypoint I do not mean `*.include.sh` is executed directly.

I mean it is sort of the API for external scripts to interact with the the birdhouse platform.

All external scripts will source this `.include.sh` file before doing pretty much anything.  So the location of this file should not change.  Also the location should be front and square so it is visible, like the `pavics-compose.sh` script.

Reply:
From the top of my head, we have external scripts for batch creating and deleting demo users, backup and sync data between prod and staging system ...

I think there are no limits to which external script are allowed to read `read-configs.include.sh`.  The backup and sync script for example, needs to access `MAGPIE_PERSIST_DIR` which is a delayed eval var to sync the DB to the staging system.

Many vars are useful to many external scripts.

Reply:
This is unrelated to this PR, but is there a reason why those scripts haven't been added to this repo? It sounds like those would all be generally useful for anyone managing a birdhouse deployment.

Reply:
These are quick scripts with lots of hardcode with hostnames and paths specific to us.  If I have time I can "generalize" those.  The easiest to generalize is probably our batch demo users creation and deletion script.

Reply:
I'm curious about your batch create/delete user script. What does it do that https://github.com/bird-house/birdhouse-deploy/blob/master/birdhouse/scripts/create-magpie-users cannot handle already? The backup/sync between systems resembles https://github.com/bird-house/birdhouse-deploy/blob/master/birdhouse/scripts/sync-data. 
Do your scripts perform other operations? Seems like nice additions to the repo if they add more capabilities.

Reply:
My batch user creation/deletion use `create-magpie-users` internally but will generate the magpie `config.yml` file with the numbers of demo users we need to bad create with some preset (email start with BOGUS_xxx@example.com and all have the same password).

For the backup/sync, `sync-data` is a full live sync, I wrote it when our new production machine went online.  For incremental sync to staging, we had a daily backup of magpie DB that we simply replicate to the staging machine.  We do not replicate the content of user jupyter data because that's big for nothing and we not the Geoserver data either because it rarely changes.  So basically aiming for speed and we don't care if data is not complete, not up-to-date.  Needs are different.